src/helpers/__tests__/global-overrides.test.js

```diff
@@ -0,0 +1,138 @@
+import type {OverrideObjectT} from '../overrides.js';
+import ThemeProvider from '../../styles/theme-provider.js';
+import BaseProvider from '../base-provider.js';
+
+type withOverridesT = {
+  overrides: OverrideObjectT,
+  forwardedRef?: React.Ref<any>,
+};
+
+describe('Global Overrides', () => {
+  test('passes global overrides through ThemeProvider', () => {
+    // Mock behavior of baseui component
+    const TestElement = (props: withOverridesT) => {
+      return (
+        <div {...props.overrides.props} style={props.overrides.style}></div>
+      );
+    };
+    const TestComponent = withOverrides(TestElement, 'TestComponent');
+    const {container} = render(
+      // $FlowFixMe
+      <ThemeProvider
+        overrides={{
+          TestComponent: {
+            props: {
+              'data-testid': 'global-overrides',
+            },
+            style: {
+              color: 'red',
+            },
+          },
+        }}
+      >
+        {/* $FlowFixMe */}
+        <TestComponent />
+      </ThemeProvider>,
+    );
+    const testComponent = getByTestId(container, 'global-overrides');
+    expect(testComponent).toBeTruthy();
+    expect(testComponent).toHaveStyle('color: red');
+  });
+  test('passes global overrides through BaseProvider', () => {
+    // Mock behavior of baseui component
+    const TestElement = (props: withOverridesT) => {
+      return (
+        <div {...props.overrides.props} style={props.overrides.style}></div>
+      );
+    };
+    const TestComponent = withOverrides(TestElement, 'TestComponent');
+    const {container} = render(
+      // $FlowFixMe
+      <BaseProvider
+        overrides={{
+          TestComponent: {
+            props: {
+              'data-testid': 'global-overrides',
+            },
+            style: {
+              color: 'red',
+            },
+          },
+        }}
+      >
+        {/* $FlowFixMe */}
+        <TestComponent />
+      </BaseProvider>,
+    );
+    const testComponent = getByTestId(container, 'global-overrides');
+    expect(testComponent).toBeTruthy();
+    expect(testComponent).toHaveStyle('color: red');
+  });
+  test('passes ref successfully', () => {
+    /** Create nested forwardRefs */
+    const TestElement = (props: {
+      overrides?: OverrideObjectT,
+      forwardedRef?: React.Ref<any>,
+    }) => {
+      return <input ref={props.forwardedRef}></input>;
+    };
+    const ForwardingElement = React.forwardRef<withOverridesT, HTMLElement>(
+      (props: withOverridesT, ref) => (
+        <TestElement {...props} forwardedRef={ref} />
+      ),
+    );
+    const TestComponent = withOverrides(ForwardingElement, 'ForwardingElement');
+    const ToRender = ({mockFn}: any) => {
+      const ref = React.createRef();
+      const handleClick = () => {
+        if (!ref.current) return;
+        mockFn();
+        ref.current.textContent = 'called';
+      };
+      return (
+        // $FlowFixMe
+        <ThemeProvider
+          overrides={{
```

Comment:
What do you think about making the overrides a property of the theme instead of a prop passed to the ThemeProvider? That seems a bit more in line with the idea of the theme as a value that you can swap around -- you might have different overrides for the light theme vs the dark theme for example.

Reply:
>  you might have different overrides for the light theme vs the dark theme for example

I think the preferred way is using theme color values in your global overrides to do light vs dark.